### PR TITLE
Ensure files are compiled as part of CI

### PR DESCRIPTION
**Pull request by sethvargo**

### CODE REVIEW

File: .github/workflows/test.yaml [hunk 1]
```diff
@@ -8,9 +8,40 @@ on:
     branches:
     - 'main'
 
+concurrency:
+  group: ${{ github.head_ref }}
+  cancel-in-progress: true
```
Comment (sethvargo) [inline .github/workflows/test.yaml:13]: This is a cool new feature I found in the docs. It automatically cancels existing runs when new commits are pushed to a PR.

**Comment by sethvargo** [comment]
Part of https://github.com/google-github-actions/auth/issues/39

@bharathkkb lmk if this is what you were thinking

**Review by bharathkkb — APPROVED** [review]
Thanks for adding this!

File: .github/workflows/test.yaml (hunk 1 above)
Comment (bharathkkb) [reply .github/workflows/test.yaml:13]: There used to be a workaround action that did this, nice to have native support 🚀

**Pull request merged**